channel/params.go

```diff
@@ -118,6 +119,11 @@ func ValidateParameters(challengeDuration uint64, numParts int, app App, nonce N
 	if len(nonce.Bytes()) > MaxNonceLen {
 		return errors.Errorf("nonce too long (%d > %d)", len(nonce.Bytes()), MaxNonceLen)
 	}
+	for i, part := range parts {
+		if part == nil {
```

Comment:
There are some particularities about checking interfaces against nil. Can you justify that doing that is ok here?

https://stackoverflow.com/questions/13476349/check-for-nil-and-nil-interface-in-go

Reply:
I did not think about this, looks like `(wallet.Address)nil` would not be detected as `nil` correctly.  
Will look into it :smile:

Reply:
Maybe the correct way of dealing with this would be to ensure that we never construct Params that are invalid?

Reply:
We do have `NewParamsUnsafe` which possibly does that.  
Otherwise when decoding `Params`, we do not check for validity of the decoded values.  

I will add a test with reflection to check for nil, since golang does not seem to offer any other possibility of checking that.

Reply:
It is advised to avoid using package reflect when possible and I think this case is not important enough to use it here.

In fact, it is currently unclear to me why we need to check for this at all? Isn't it the problem of the framework user if it sets a participant address to nil? Why would we need to protect against this?

Reply:
@ggwpez hm? can you explain where exactly you were running in problems if you don't check for nil participants? Otherwise I would close the issue.

Reply:
Well, the problem that i tried to fix is described in #69. But it seems not possible without the use reflection. @matthiasgeihs

Reply:
#69 only describes a potential problem, but I have trouble understanding if this is an actual problem or it is rather artificial. This is why I wrote above:

> In fact, it is currently unclear to me why we need to check for this at all? Isn't it the problem of the framework user if it sets a participant address to nil? Why would we need to protect against this?

Reply:
It is something that i noticed when changing code for the hackathon.  
I dont know if there is currently a problem with it, i just thought that it is weird if `ValidateParams` does not actually validate the params.

Reply:
OK, so it seems that there is not an actual problem with it for now?

I am not super convinced that we need it, especially since it would be the first usage of package reflect.

Reply:
Well ok then lets wait for it to pops up somewhere again.